### ISSUE TITLE
macros (as in Jinja)

### ISSUE THREAD

**Issue by snej** [pantor/inja#238]
The [macro](https://jinja.palletsprojects.com/en/3.0.x/templates/#macros) feature in Jinja would be super great to have. I'm finding myself repeating lots of small snippets in my templates, things for which `include` in't well suited (too heavyweight and doesn't take parameters.)

**Comment by xahon** [comment]
@snej Inlcude takes variables from current scope but it can't set output variables, unfortunately

**Comment by navrocky** [comment]
I agree, it's a much-needed feature. Currently, I'm using the includes as workaround, but it's inconvenient.
